lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -1117,7 +1117,9 @@ std::vector<SecondarySkill> CGHeroInstance::getLevelUpProposedSecondarySkills()
 	std::vector<SecondarySkill> obligatorySkills; //hero is offered magic school or wisdom if possible
 	if (!skillsInfo.wisdomCounter)
 	{
-		if (cb->isAllowed(2, SecondarySkill::WISDOM) && !getSecSkillLevel(SecondarySkill::WISDOM))
+		if (cb->isAllowed(2, SecondarySkill::WISDOM) &&
```

Comment:
I see in all 3 cases we use both isAllowed and heroClass together. Just an idea, what if we create an overload for isAllowed and pass hero instance iinside?

Reply:
Yeah, makes sense. Although probably not override but method of Hero class, similar to existing
CGHeroInstance::canLearnSpell